### PR TITLE
OSJSON and Swift Sample

### PR DESCRIPTION
**Pull request by drhaynes**
This PR:
- Updates the parsing & decoding code to use OSJSON.
- Removes all other JSON parsing/decoding frameworks.
- Adds a Swift example app.

### CODE REVIEW

File: Search/Header.swift [hunk 1]
```diff
@@ -1,16 +1,16 @@
 //
 //  Header.swift
-//  Search
```
Comment (dhardiman) [inline Search/Header.swift:3]: Hmm, should probably put the module name in to the templates. I'll do that this morning.

File: Search/Header.swift [hunk 2]
```diff
@@ -37,34 +37,51 @@ public class Header: NSObject {
...
+            return nil
         }
+        let matchprecision = json.intValueForKey(Header.MatchprecisionKey)
+        let maxresults = json.intValueForKey(Header.MaxresultsKey)
+        let totalresults = json.intValueForKey(Header.TotalresultsKey)
+        let offset = json.intValueForKey(Header.OffsetKey)
 
         self.init(
-            matchprecision: Int(matchPrecision),
+            matchprecision: Int(matchprecision),
```
Comment (dhardiman) [inline Search/Header.swift]: Also, probably don't need this any more do we? Might be able to remove this in the template. And get rid of the trailing white space. There is a setting in Xcode to get it to trim that for you though.

File: Search/Response+Parsing.swift [hunk 3]
```diff
@@ -39,28 +39,31 @@ private func parseResponse(data: NSData?) -> Result<Response> {
...
         return .Failure(SearchError.FailedToDeserialiseJSON)
     }
     return .Success(response)
 }
 
-private func responseFromJson(json: JSON) -> Response? {
-    let headerJson = json[Response.HeaderKey]
-    guard let header = Header(json: headerJson) else {
-        return nil
+private func responseFromOSJson(json: OSJSON) -> Response? {
```
Comment (dhardiman) [inline Search/Response+Parsing.swift]: This could just use the failable initialiser pattern on `Response`.

Comment (dhardiman) [inline Search/Response+Parsing.swift]: In fact the template has a version of this already, and then you can bin the `dpaFromOSJson` function as well.

File: Search/SearchService.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+//  Search
+//
+//  Created by David Haynes on 10/03/2016.
+//  Copyright © 2016 Ordnance Survey. All rights reserved.
+//
+
+import Fetch
+
+/// Service for consuming the OS Places API
+public class SearchService: Searchable {
```
Comment (dhardiman) [inline Search/SearchService.swift]: Wonder whether the name of this class should be more specific about searching OS Places? To allow for some of the other possible sources of data as the Android framework is including.

File: SearchExampleSwift/APIKEY [hunk 5]
```diff
@@ -0,0 +1 @@
+kla5ReI1XnDzOtqJoYcIxtSseCEgNtaf
```
Comment (dhardiman) [inline SearchExampleSwift/APIKEY]: Committed key. Remove this and squash the commit where it was added so it's gone from history...

File: SearchExampleSwift/ResultsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    override func viewDidLoad() {
+        let apiKey = NSBundle.mainBundle().URLForResource("APIKEY", withExtension: nil).flatMap { url -> String? in
+            do {
+                return try String(contentsOfURL: url)
+            } catch {
+                return nil
+            }
+        } ?? ""
+
+        self.searchService = SearchService(apiKey: apiKey)
```
Comment (dhardiman) [inline SearchExampleSwift/ResultsViewController.swift]: Lots of unnecessary use of `self.` in this class.

File: SearchExampleSwift/ResultsViewController.swift [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+        super.viewDidLoad()
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+
+    internal func search(text: String) {
+        if(text.utf16.count > 2) {
```
Comment (dhardiman) [inline SearchExampleSwift/ResultsViewController.swift]: No need for parens in swift `if`

File: SearchExampleSwift/ResultsViewController.swift (hunk 6 above)
Comment (drhaynes) [inline SearchExampleSwift/ResultsViewController.swift]: Yeah good spot, had this in an `init` and got left when I switched back to `viewDidLoad`

Other use is inside a closure.

File: SearchExampleSwift/ResultsViewController.swift (hunk 7 above)
Comment (drhaynes) [inline SearchExampleSwift/ResultsViewController.swift]: Indeed, will update.

**Comment by drhaynes** [comment]
Updated

**Comment by drhaynes** [comment]
Tests will pass once https://github.com/OrdnanceSurvey/osjson-ios/pull/9 is merged in OSJSON.

**Comment by dhardiman** [comment]
Other pr is merged, so happy with this once updated and tests pass.

**Pull request merged**